.travis/before_install.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+set -x
+
+# Travis-CI before_install
+
+# Clear bashrc - the default does nothing if not in an interactive shell.
+# SSH connections use the ~/.bashrc file for their environment, so we'll be
+# loading our python environment here.
+
+echo > ~/.bashrc
+
+# Setup virtualenv (using system packages for pygtk as pip won't install it)
+
+virtualenv --system-site-packages $HOME/virtualenv/cylc
```

Comment:
Oh, you are correct @oliver-sanders ! Had a look at the Travis-CI documentation, and it already creates a virtualenv by default.

I guess the extra virtualenv was being created for system packages, as by default the virtualenv that Travis-CI creates has no access to packages installed via apt-get, for example. So pygtk wouldn't work.

There is an option in Travis-CI, `virtualenv / system_site_packages`, which changes this behaviour. This will reduce one virtualenv :-) so our build should speed up a bit more.

The only remaining part in `before_install.sh` was the `PATH` variable, that needs to include the Cylc `bin` directory. Fixed that by creating a `env / global` value, that will be set once for all stages/jobs/etcs 👍